hfile.c

```diff
@@ -735,12 +778,38 @@ static hFILE *hopen_mem(const char *url, const char *mode)
         hts_decode_percent(buffer, &length, data);
     }
 
-    hFILE_mem *fp = (hFILE_mem *)
-        hfile_init_fixed(sizeof (hFILE_mem), mode, buffer, length, size);
-    if (fp == NULL) { free(buffer); return NULL; }
+    return create_hfile_mem(buffer, mode, length, size);
```

Comment:
You still need to free `buffer`. I would check the value returned by `create_hfile_mem` and, if `NULL`, `free(buffer)`.